src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -198,14 +199,254 @@ internal static bool AreEqualCollections<T>(ICollection<T> x, ICollection<T> y,
             return current;
         }
 
+        /// <summary>
+        /// Try to find a target header value among the set of given header values and parse it as a
+        /// <see cref="TimeSpan"/>.
+        /// </summary>
+        /// <param name="headerValues">
+        /// The <see cref="StringValues"/> containing the set of header values to search.
+        /// </param>
+        /// <param name="targetValue">
+        /// The target header value to look for.
+        /// </param>
+        /// <param name="value">
+        /// When this method returns, contains the parsed <see cref="TimeSpan"/>, if the parsing succeeded, or
+        /// null if the parsing failed. The conversion fails if the <paramref name="targetValue"/> was not
+        /// found or could not be parsed as a <see cref="TimeSpan"/>. This parameter is passed uninitialized;
+        /// any value originally supplied in result will be overwritten.
+        /// </param>
+        /// <returns>
+        /// <code>true</code> if <paramref name="targetValue"/> is found and successfully parsed; otherwise,
+        /// <code>false</code>.
+        /// </returns>
+        // e.g. { "headerValue=10, targetHeaderValue=30" }
+        public static bool TryParseTimeSpan(StringValues headerValues, string targetValue, out TimeSpan? value)
```

Comment:
This name is misleading, you're not parsing a fully formatted time, only integer seconds. You just happen to represent the output as a TimeSpan. TryParseSeconds?